test/e2e/storage/utils/host_exec.go

```diff
@@ -84,29 +105,63 @@ func (h *hostExecutor) launchNodeExecPod(node string) *v1.Pod {
 	return pod
 }
 
-// IssueCommandWithResult issues command on given node and returns stdout.
-func (h *hostExecutor) IssueCommandWithResult(cmd string, node *v1.Node) (string, error) {
+// Execute executes the command on the given node. If there is no error
+// performing the remote command execution, the stdout, stderr and exit code
+// are returned.
+// This works like ssh.SSH(...) utility.
+func (h *hostExecutor) Execute(cmd string, node *v1.Node) (Result, error) {
+	result, err := h.exec(cmd, node)
+	if codeExitErr, ok := err.(exec.CodeExitError); ok {
+		// extract the exit code of remote command and silence the command
+		// non-zero exit code error
+		result.Code = codeExitErr.ExitStatus()
+		err = nil
+	}
+	return result, err
+}
+
+func (h *hostExecutor) exec(cmd string, node *v1.Node) (Result, error) {
+	result := Result{
+		Host: node.Name,
+		Cmd:  cmd,
+	}
 	pod, ok := h.nodeExecPods[node.Name]
```

Comment:
I remember @jsafrane hit some issue with creating two host exec pods in the same test. Did that ever get resolved?  How about hostexec for two tests running in parallel?

Reply:
It had been resolved. Originally, it used a static pod name. If a test has two instances of HostExec, pod names will conflict. @jsafrane switched to use `GenerateName` to avoid pod name conflict [here](https://github.com/kubernetes/kubernetes/pull/83108#issuecomment-535452553).

Reply:
A limitation is this utility cannot be shared by multiple parallel tests. We can add support when necessary.